src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.ForEachAsync.cs

```diff
@@ -92,21 +92,12 @@ private static Task ForAsync<T>(T fromInclusive, T toExclusive, int dop, TaskSch
                 return Task.CompletedTask;
             }
 
+#pragma warning disable CS8500
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             static bool Interlockable() =>
-                typeof(T) == typeof(sbyte) ||
-                typeof(T) == typeof(byte) ||
-                typeof(T) == typeof(short) ||
-                typeof(T) == typeof(ushort) ||
-                typeof(T) == typeof(char) ||
-                typeof(T) == typeof(int) ||
-                typeof(T) == typeof(uint) ||
-                typeof(T) == typeof(long) ||
-                typeof(T) == typeof(ulong) ||
-                typeof(T) == typeof(nint) ||
-                typeof(T) == typeof(nuint);
+                !RuntimeHelpers.IsReferenceOrContainsReferences<T>() && RuntimeHelpers.IsBitwiseEquatable<T>() &&
```

Comment:
IsBitwiseEquatable is internal to corelib.

Reply:
Ah right, forgot Parallel isn't in CoreLib. I guess it doesn't make much sense to move it there and it's better to close this and wait for `IsBitwiseEquatable` to become public?

Reply:
> it's better to close this

Yup. Thanks, anyway.